src/EventHub/EventHub/Cmdlets/NetworkRuleSet/SetAzureEventHubNetworkrule.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Azure.Commands.EventHub.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.NetworkruleSet
+{
+    /// <summary>
+    /// 'Set-AzEventHubNamespace' Cmdlet updates the specified Eventhub Namespace
+    /// </summary>
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventHubNetworkRuleSet", SupportsShouldProcess = true, DefaultParameterSetName = NetwrokruleSetPropertiesParameterSet), OutputType(typeof(PSNetworkRuleSetAttributes))]
+    public class SetAzureEventHubNetworkrule : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }       
+
+        [Parameter(Mandatory = false, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "Default Action for NetwrokeuleSet")]
+        [ValidateSet("Allow", "Deny", IgnoreCase = true)]
+        [PSArgumentCompleter("Allow", "Deny")]
+        [PSDefaultValue(Value ="Deny")]
+        public string DefaultAction { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 3, HelpMessage = "List of IPRuleSet")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNWRuleSetIpRulesAttributes> IPRules { get; set; }
```

Comment:
@v-Ajnava parameters should be arrays instead of lists (this provides a better viewing experience in the docs with no impact on the cmdlet itself)

This comment applies to all cmdlets in this PR

Reply:
Resolved